xlators/mgmt/glusterd/src/glusterd.h

```diff
@@ -248,8 +248,8 @@ typedef struct glusterd_add_dict_args {
 
 typedef struct glusterd_friend_synctask_args {
     dict_t *peer_data;
-    dict_t *peer_ver_data;  // Dictionary to save peer version data
-    uint64_t *status_arr;   // Array to save volume update status
+    dict_t *peer_ver_data;   // Dictionary to save peer version data
+    uint64_t status_arr[1];  // Array to save volume update status
```

Comment:
I would explicitly say that the real size of the array is dynamically allocated based on the number of volumes.